src/GraphQL/Introspection.hs

```diff
@@ -0,0 +1,317 @@
+typeDefinitionToAST (TypeDefinitionScalar s)        = AST.TypeDefinitionScalar $ scalarTypeDefinitionToAST s
+typeDefinitionToAST (TypeDefinitionEnum e)          = AST.TypeDefinitionEnum $ enumTypeDefinitionToAST e
+typeDefinitionToAST (TypeDefinitionTypeExtension e) = AST.TypeDefinitionTypeExtension $ extensionTypeDefinitionToAST e
+
+objectTypeDefinitionToAST :: ObjectTypeDefinition -> AST.ObjectTypeDefinition
+objectTypeDefinitionToAST (ObjectTypeDefinition name interfaces fields) =
+  AST.ObjectTypeDefinition name (map (AST.NamedType . getName) interfaces) (NonEmpty.toList $ map fieldDefinitionToAST fields)
+
+inputObjectTypeDefinitionToAST :: InputObjectTypeDefinition -> AST.InputObjectTypeDefinition
+inputObjectTypeDefinitionToAST (InputObjectTypeDefinition name fields) =
+  AST.InputObjectTypeDefinition name (NonEmpty.toList $ map inputObjectFieldDefinitionToAST fields)
+
+interfaceTypeDefinitionToAST :: InterfaceTypeDefinition -> AST.InterfaceTypeDefinition
+interfaceTypeDefinitionToAST (InterfaceTypeDefinition name fields) =
+  AST.InterfaceTypeDefinition name (NonEmpty.toList $ map fieldDefinitionToAST fields)
+
+unionTypeDefinitionToAST :: UnionTypeDefinition -> AST.UnionTypeDefinition
+unionTypeDefinitionToAST (UnionTypeDefinition name objects) =
+  AST.UnionTypeDefinition name (NonEmpty.toList $ map (AST.NamedType . getName) objects)
+
+scalarTypeDefinitionToAST :: ScalarTypeDefinition -> AST.ScalarTypeDefinition
+scalarTypeDefinitionToAST (ScalarTypeDefinition name) =
+  AST.ScalarTypeDefinition name
+
+enumTypeDefinitionToAST :: EnumTypeDefinition -> AST.EnumTypeDefinition
+enumTypeDefinitionToAST (EnumTypeDefinition name values) =
+  AST.EnumTypeDefinition name $ map (AST.EnumValueDefinition . getName) values
+
+extensionTypeDefinitionToAST :: TypeExtensionDefinition -> AST.TypeExtensionDefinition
+extensionTypeDefinitionToAST (TypeExtensionDefinition obj) =
+  AST.TypeExtensionDefinition $ objectTypeDefinitionToAST obj
+
+fieldDefinitionToAST :: FieldDefinition -> AST.FieldDefinition
+fieldDefinitionToAST (FieldDefinition name args out) =
+  AST.FieldDefinition name (map argToInputValue args) (typeToAST out)
+
+argToInputValue :: ArgumentDefinition -> AST.InputValueDefinition
+argToInputValue (ArgumentDefinition name annotatedInput _) =
+  AST.InputValueDefinition name (inputTypeToAST annotatedInput) Nothing -- FIXME
+
+inputObjectFieldDefinitionToAST :: InputObjectFieldDefinition -> AST.InputValueDefinition
+inputObjectFieldDefinitionToAST (InputObjectFieldDefinition name annotatedInput _) = AST.InputValueDefinition name (inputTypeToAST annotatedInput) Nothing -- FIXME
+
+typeToAST :: AnnotatedType GType -> AST.GType
+typeToAST (TypeNamed t) =
+  -- AST.TypeNamed $ AST.NamedType $ getName t
+  AST.TypeNonNull $ AST.NonNullTypeNamed $ AST.NamedType $ getName t
+typeToAST (TypeList (ListType t)) =
+  -- AST.TypeList $ AST.ListType $ AST.TypeNamed $ AST.NamedType $ getName t
```

Comment:
do you still need the commented out lines? If so maybe add a `TODO` saying what needs to be done to uncomment them?